devbuild.sh

```diff
@@ -73,20 +73,17 @@ cd ${BUILD_DIR}
 cmake .. -DCMAKE_INSTALL_PREFIX=..
 make -j ${BUILD_JOBS:-4}
 
-. ${MYDIR}/${ENV_FILE}_DA
-cd ${BUILD_DIR}
-mkdir -p build_gsi
-cd build_gsi
-cmake ${MYDIR}/src/gsi -DBUILD_CORELIBS=ON -DCMAKE_INSTALL_PREFIX=.
-make -j ${BUILD_JOBS:-4}
-cp ./bin/gsi.x ${MYDIR}/bin/.
+cd ${MYDIR}/src/gsi
+./ush/build.comgsi
+cp ${MYDIR}/src/gsi/build/bin/gsi.x ${MYDIR}/bin/gsi.x
```

Comment:
Why are we going away from a cmake build here? How is the environment managed?

Reply:
It is intent to let the GSI manage its own compile. The APP only set the run environment now.

Reply:
I know I'm not a selected reviewer, but I get notifications and saw this come through.  I know the intent here is to "make things work", and I understand that goal and deadline driven activities.  But speaking as someone who has spent/wasted countless hours, days, weeks? wrestling with non portable build systems, trying to make them work in ways the developers did not foresee, this is the type of thing that makes life truly miserable for others.  I can say without hesitation that the lack of a proper, portable, build system is the root cause of major delays in delivering on project goals for work I've been involved with.  Compliance with the existing `cmake` build design should be a **minimum** barrier of entry for anything that goes into any part of the UFS.  Adding stuff like this takes the UFS backwards, not forwards.  I realize that what I'm asking is a heavy lift, because so many parts we want to add do not yet have a functioning `cmake` build system.  But, the price we pay when we do not adhere to standards is that users cannot do their work because they spend months trying to figure out how to work around all of the machine-dependent idiosyncrasies.  Even the existing `cmake` build still has many issues in that regard, but it is at least moving in the right direction.

Please consider what you are trading off by adding things like this "to make it work" at the expense of the standards that need to be upheld and the resulting cost to developers and users down the road. And also beware of the inertia of stuff that goes into a repository.  Once it goes in, you can't take it out as easily.  And bad things that go in, often stay bad for a very long time.  If this change is rejected, the motivation to fix it high.  If it is accepted, the motivation to do it correctly is non-existent, and will likely not happen for a very long time.

Reply:
@christinaholtNOAA I don't think the way I am using will cause any trouble in the future. It follows idea of separate concern. It is not UFS app's task to make GSI or other components portable or work on every platform. It is GSI developer's task.  Here, GSI has its own system to make it works on all the platforms we need. We don't need to copy everything from GSI's compiling system to UFS and maintain it in the future. If anything wrong with GSI, the GSI developer should fix it, not the UFS APP developer.

Reply:
It is the job of the UFS Apps to build all the components that it needs to run the application. It is an effort of UFS to have all those components build with CMake. Please do not undo the CMake build that was already implemented here. This change is going in the opposite direction of what we need to be accomplishing with this build.

Reply:
I agree with @christinaholtNOAA here.  Removing the `cmake` build makes no sense to me and appears to be a regression in features/capability as it replaces a standard portable build mechanism with an ad-hoc one.  There is going in the wrong direction.  What should be happening instead is to improve upon the existing CMake such that any given package's build system does not build other external packages but uses Cmake's `find_package` utility to discover dependencies and fail if they are not found.